app/code/community/Reverb/ProcessQueue/Helper/Task.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Author: Sean Dunagan (https://github.com/dunagan5887)
+ * Created: 12/9/15
+ */
+
+class Reverb_ProcessQueue_Helper_Task extends Mage_Core_Helper_Data
+{
+    const STALE_TIME_LAPSE = '-2 weeks';
+
+    /**
+     * The calling block is expected to catch exceptions
+     *
+     * @param null|string $task_code
+     * @return int - Number of rows deleted
+     */
+    public function deleteStaleSuccessfulTasks($task_code = null)
+    {
+        $current_gmt_timestamp = Mage::getSingleton('core/date')->gmtTimestamp();
+        $stale_timestamp = strtotime(self::STALE_TIME_LAPSE, $current_gmt_timestamp);
+        $stale_date = date('Y-m-d H:i:s', $stale_timestamp);
+
+        $rows_deleted = Mage::getResourceSingleton('reverb_process_queue/task')
+                            ->deleteSuccessfulTasks($task_code, $stale_date);
```

Comment:
## 2015-12-10 22:20:35
## 29107 ## QUERY
SQL: DELETE FROM `reverb_process_queue_task` WHERE (status=3) AND (last_executed_at < '2015-11-26 22:20:34')
AFF: 10119
TIME: 0.0878

## 2015-12-10 22:20:37
## 29107 ## QUERY
SQL: DELETE FROM `reverb_process_queue_task_unique` WHERE (status=3) AND (last_executed_at < '2015-11-26 22:20:34')
AFF: 0
TIME: 0.0007

I'm working on an Ubuntu VM living in a Macbook Retina with 8GB RAM, quad-core and SSD though, so the servers of smaller vendors likely won't match my performance. Nonetheless, I can't see their production servers operating any more than 10x slower than mine, which still leaves 10,000 at around a second